io/csv/include/io/csv_io.hpp

```diff
@@ -56,13 +58,14 @@ namespace detray
                     const std::string &layer_volume_file_name,
                     const std::string &grid_file_name,
                     const std::string &grid_entries_file_name,
+		    vecmem::memory_resource& resource,
                     scalar r_sync_tolerance = 0.,
                     scalar z_sync_tolerance = 0.)
   {
 
     using typed_detector = detector<array_type, tuple_type, vector_type, alignable_store, surface_source_link, bounds_source_link>;
 
-    typed_detector d(detector_name);
+    typed_detector d(detector_name, resource);
```

Comment:
The resource should be passed directly to the grid, I think

Reply:
This was for initializing `volume_grid` (member of detector class) with memory resource during the constructor. From my experience, It is not possible to use vecmem memory structure without initializing a vector with memory resource (@krasznaa is it correct?)

Reply:
But we cannot use the same memory resource for the grid and, say, the transform store? Maybe I just don't understand what its doing

Reply:
But we cannot use the same memory resource for the grid and, say, the transform store? Maybe I just don't understand what its doing

Reply:
:confused: I don't think I fully understand the code here...

You can absolutely use the same memory resource in many places. In fact, you should only instantiate the minimum number of memory resource objects that your application needs, and then just use those same instances everywhere.

With VecMem we do try to make client code always explicitly declare what memory resource it wants to use for a given container. So it is true, that if at all technically feasible, every `vecmem::` "host" container should always be constructed with an explicit memory resource. Which means that we have to pass these references/pointers around in our code.

How the memory resource reference/pointer gets to its "final destination", is a matter of code design. Having the `detector` class receive a memory resource explicitly is an okay way to construct it on the host. Though remember that on a device we will need to construct the `detector` object from vector views. So we'll need some specific constructor for that as well... (It is okay in my opinion to have different constructors for the host and the device code, that's not a bad design by itself.)

Reply:
Ah, thank you. I was just wondering, because the detector keeps up to 17 containers besides the grid right now, that we will have to handle at some point. Potentially more, if we consider alignment, I think.

Reply:
> Though remember that on a device we will need to construct the detector object from vector views

Of course yes, but it's not in scope of this PR. This will be follow-up in the later PRs